xbmc/guilib/StereoscopicsManager.cpp

```diff
@@ -509,6 +509,8 @@ void CStereoscopicsManager::OnPlaybackStarted(void)
   STEREOSCOPIC_PLAYBACK_MODE playbackMode = (STEREOSCOPIC_PLAYBACK_MODE) CSettings::GetInstance().GetInt(CSettings::SETTING_VIDEOPLAYER_STEREOSCOPICPLAYBACKMODE);
   RENDER_STEREO_MODE mode = GetStereoMode();
 
+  g_infoManager.UpdateAVInfo();
```

Comment:
if there is a way to directly ask the video player if the currently played file is 3D, then that would ofc be the better solution. Going the way via GUIinfoManager was the only way I had seen back then, and Infomanager had that info anyways for skin purposes.
Otoh, the 3D toggle logic should be implemented differently anyways, like in a way that won't in worst case trigger two resolution switches in a row. This however was (is) waaaaaay beyond my knowledge and skills and I didn't want to somewhat UI related code to the dvdplayer back then, which is why I went the way with messages.